apps/oxlint/src/lint.rs

```diff
@@ -381,20 +407,73 @@ impl Runner for LintRunner {
                 let mut stdin = child.stdin.take().expect("Failed to open tsgolint stdin");
 
                 std::thread::spawn(move || {
-                    // TODO: Implement JSON input.
-                    let json = serde_json::to_string(&"{}").expect("Failed to serialize JSON");
+                    let json =
+                        serde_json::to_string(&json_input).expect("Failed to serialize JSON");
 
                     stdin.write_all(json.as_bytes()).expect("Failed to write to tsgolint stdin");
                 });
 
-                // TODO: Implement output parsing.
-                child.wait_with_output().expect("Failed to wait for tsgolint process");
+                // TODO: Stream diagnostics as they are emitted, rather than waiting
+                let output = child.wait_with_output().expect("Failed to wait for tsgolint process");
+
+                match parse_tsgolint_output(&output.stdout) {
+                    Ok(parsed) => {
+                        let oxc_diagnostics = parsed
+                            .into_iter()
+                            .map(|tsgolint_diagnostic| {
+                                (tsgolint_diagnostic.file_path.clone(), tsgolint_diagnostic.into())
+                            })
+                            // group diagnostics by file path
+                            .fold(
+                                FxHashMap::default(),
+                                |mut acc: FxHashMap<PathBuf, Vec<OxcDiagnostic>>,
+                                 (file_path, diagnostic)| {
+                                    acc.entry(file_path).or_default().push(diagnostic);
+
+                                    acc
+                                },
+                            )
+                            .into_iter()
+                            .map(|(file_path, diagnostics)| {
+                                (
+                                    file_path.clone(),
+                                    DiagnosticService::wrap_diagnostics(
+                                        tsgolint_state.cwd.clone(),
+                                        file_path.clone(),
+                                        &read_to_string(&file_path)
```

Comment:
note: this is probably not a good idea. we shouldn't re-read every file just for diagnostics. maybe we should add a file source text cache somewhere?

Reply:
🤔 it's not going to be a huge problem if error count is low.